### ISSUE TITLE
gumbel-softmax in fairseq

### ISSUE THREAD

**Issue by nicolabertoldi** [pytorch/fairseq#1207]
is the gumbel trick implemented in fairseq?

**Comment by myleott** [comment]
Not currently, but PRs welcome :)
